sdcm/utils/common.py

```diff
@@ -992,6 +997,82 @@ def clean_load_balancers_aws(tags_dict, dry_run=False):
                     LOGGER.debug("Failed with: %s", str(ex))
 
 
+def list_cloudformation_stacks_aws(tags_dict=None, region_name=None, group_as_region=False, verbose=False):
+    """
+    list all cloudformation stacks with specific tags AWS
+
+    :param tags_dict: a dict of the tag to select the instances, e.x. {"TestId": "9bc6879f-b1ef-47e1-99ab-020810aedbcc"}
+    :param region_name: name of the region to list
+    :param group_as_region: if True the results would be grouped into regions
+    :param verbose: if True will log progress information
+
+    :return: cloudformation staacks dict where region is a key
+    """
+    cloudformation_stacks = {}
+    aws_regions = [region_name] if region_name else all_aws_regions()
+
+    def get_stacks(region):
+        if verbose:
+            LOGGER.info('Going to list aws region "%s"', region)
+        time.sleep(random.random())
+        tagging = boto3.client('resourcegroupstaggingapi', region_name=region)
+        paginator = tagging.get_paginator('get_resources')
+        tag_filter = [{'Key': key, 'Values': [value]}
+                      for key, value in tags_dict.items() if key != 'NodeType']
+
+        tag_mappings = itertools.chain.from_iterable(
+            page['ResourceTagMappingList']
+            for page in paginator.paginate(TagFilters=tag_filter, ResourceTypeFilters=['cloudformation:stack'])
+        )
+        cloudformation_stacks[region] = list(tag_mappings)
+        if verbose:
+            LOGGER.info("%s: done [%s/%s]", region, len(list(cloudformation_stacks.keys())), len(aws_regions))
+
+    ParallelObject(aws_regions, timeout=100).run(get_stacks, ignore_exceptions=False)
```

Comment:
For the moment we have 16 elements in the `aws_regions` list.
By default, ParallelObject, will run everything with concurrency `6`.
So, I think you may want to provide here additional option -> `num_workers=len(aws_regions)`.

Reply:
this is a change we probably want across all of those functions, not related to this specific one.

Reply:
changed it